File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcFurnishingElementType/IfcFurnitureType/IfcFurnitureType.cs
public class IfcFurnitureType : IfcFurnishingElementType
{
    public IfcAssemblyPlaceEnum AssemblyPlace { get; set; }
    public IfcFurnitureTypeEnum PredefinedType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcFurnitureTypeEnum.USERDEFINED) OR ((PredefinedType = IfcFurnitureTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcFurnishingElementType/IfcFurnitureType/IfcFurnitureType.js
class IfcFurnitureType extends IfcFurnishingElementType {
    constructor() {
        /** @type {IFCASSEMBLYPLACEENUM} */
        this.AssemblyPlace = null;
        /** @type {IFCFURNITURETYPEENUM} */
        this.PredefinedType = null;
    }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcFurnitureTypeEnum.USERDEFINED) OR ((PredefinedType = IfcFurnitureTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}
